idShiftUtils/idShiftUtils.go

```diff
@@ -57,7 +57,26 @@ type IDMapping struct {
 	Size        uint32
 }
 
-// Shifts the ACL type user and group IDs by the given offset
+// checkACLSupport attempts to set an extended ACL attribute on a file to check ACL support.
+func checkACLSupport(path string) bool {
+    file, err := os.Open(path)
+    if err != nil {
+        fmt.Println("Error opening file:", err)
+        return false
+    }
+    defer file.Close()
+
+    // Try setting an extended attribute specific to ACLs
+    err = unix.Fsetxattr(int(file.Fd()), "system.posix_acl_access", []byte{}, 0)
+    if err == unix.ENOTSUP {
+        return false
+    }
+    // Any other error indicates something else went wrong, so we assume ACLs
+    // are supported
+    return err == nil
```

Comment:
Your comment says that any other error would serve to indicate that ACLs are supported, but you're only returning "true" when there's no error at all.

Reply:
Done.